packages/tailwindcss/src/ast.ts

```diff
@@ -208,18 +208,151 @@ export function walkDepth(
   }
 }
 
-export function toCss(ast: AstNode[]) {
-  let atRoots: string = ''
+// Optimize the AST for printing where all the special nodes that require custom
+// handling are handled such that the printing is a 1-to-1 transformation.
+export function optimizeAst(ast: AstNode[]) {
```

Comment:
Hm now that I see this in the whole PR I'm not so super convinced about this name anymore. The reason being that we refer to `optimize` as the step in post-compilation where we run `lightningcss`. Now we're overloading the term with something used during the compilation. Maybe `applyFallbacksToAst` or something like this? 🤔 I also dig your initial name of `prepareAstForPrinting` honestly lol. Find it less confusing then overloading the term.

Reply:
Hmm I see. The problem is that this is doing multiple things. While `applyFallbacksToAst` is correct, we are also in addition to tracking the fallbacks making sure that `@property` is handled once, that `Context` nodes become transparent and that `AtRoot` nodes are hoisted.

`prepareAstForPrinting` is a bit more vague which is good to describe that we are doing multiple things. But the `forPrinting` part is off because in `@tailwindcss/postcss` that would be `prepareAstForTransformingIntoOtherAST` 😅